go/consensus/tendermint/apps/registry/transactions.go

```diff
@@ -217,7 +216,17 @@ func (app *registryApplication) registerNode( // nolint: gocyclo
 
 	// Charge gas for node registration if signed by entity. For node-signed
 	// registrations, the gas charges are pre-paid by the entity.
-	isEntitySigned := sigNode.MultiSigned.IsSignedBy(newNode.EntityID)
+	//
+	// Yes, the detection logic is awful.  See VerifyRegisterNodeArgs,
+	// where the same thing is done for a justification.
+	var isEntitySigned bool
+	for _, v := range sigNode.MultiSigned.Signatures {
+		addr := staking.NewAddress(multisig.NewAccountFromPublicKey(v.PublicKey))
+		if addr == untrustedNode.EntityAddress {
```

Comment:
would `newNode.EntityAddress` have worked here?

Reply:
Sure.  There's essentially no difference, but I'll use newNode for consistency.